src/enemy.asm

```diff
@@ -0,0 +1,442 @@
+;
```

Comment:
When it comes to use of global variables such as temp_1 and temp_address, all this code becomes NMI-unsafe.
In interrupt can occur after any instruction and if the interrupt handler itself modifies anything in temp_-stuff, after the execution returns to the subroutine, the data will be in unknown state and everything will be screwed.

The code should be rewritten to be context independent and use only locals on stack. This should be discussed and a generic mechanism developed.